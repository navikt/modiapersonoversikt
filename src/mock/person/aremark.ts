import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    tilrettelagtKomunikasjonsListe: [
        {
            behovKode: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    personstatus: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [{
        harSammeBosted: false,
        rolle: Relasjonstype.Barn,
        tilPerson: {
            navn: {
                sammensatt: '',
                fornavn: 'Siri',
                mellomnavn: null,
                etternavn: 'Aremark'
            },
            alder: 17,
            fødselsnummer: '10117100000',
            personstatus: {}
        }

    }],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            telefonnummer: '99887766',
            retningsnummer: '47',
            sistEndretAv: 'BRUKER'
        }
    }
};