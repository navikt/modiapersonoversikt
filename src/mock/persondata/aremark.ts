import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';
import { harDiskresjonskode } from '../../app/personside/visittkort-v2/visittkort-utils';

export function hentBarnAremark(): ForelderBarnRelasjon[] {
    return [
        {
            ident: '12345678910',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: 13,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [
                {
                    kode: AdresseBeskyttelse.UGRADERT,
                    beskrivelse: 'UGRADERT'
                }
            ],
            harSammeAdresse: false,
            dodsdato: []
        },
        {
            ident: '12345678911',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: null,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [],
            harSammeAdresse: false,
            dodsdato: []
        }
    ];
}

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    geografiskTilknytning: '0118',
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge',
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2010-02-20T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    oppholdsAdresse: [
        {
            linje1: 'Oppholdsadresse 1',
            linje2: '0000 AREMARK',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum',
        publikumsmottak: [
            {
                besoksadresse: {
                    linje1: 'Adressevei 1',
                    linje2: '0000 AREMARK',
                    linje3: null,
                    sistEndret: null,
                    gyldighetsPeriode: null
                },
                apningstider: [
                    {
                        ukedag: 'Mandag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Tirsdag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Onsdag',
                        apningstid: '10.00 - 16.00'
                    }
                ]
            }
        ]
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Kode 6'
        }
    ],
    sikkerhetstiltak: [
        {
            type: 'TFUS',
            beskrivelse: 'Telefonisk utestengelse',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate,
            sivilstandRelasjon: {
                fnr: '12345555555',
                navn: [
                    {
                        fornavn: 'BAREMARK',
                        mellomnavn: null,
                        etternavn: 'TESTFAMILIEN'
                    }
                ],
                alder: 40,
                adressebeskyttelse: [
                    {
                        kode: AdresseBeskyttelse.UGRADERT,
                        beskrivelse: 'UGRADERT'
                    }
                ],
                harSammeAdresse: true
            }
        }
    ],
    foreldreansvar: hentBarnAremark().map((barn) => ({
        ansvar: 'felles',
        ansvarlig: null,
        ansvarsubject: {
            navn: harDiskresjonskode(barn.adressebeskyttelse) ? null : barn.navn.firstOrNull(),
            ident: barn.ident
        }
    })),
    deltBosted: [
        {
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            },
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'Folkeregisteret'
            }
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omrade: [
                {
                    kode: '*',
                    beskrivelse: ''
                }
            ],
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'Voksen',
            omfang: 'Ivareta personens interesser innenfor det personlige og økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-10' as LocalDate
            }
        },
        {
            ident: '123456799',
            navn: {
                fornavn: 'Truls',
                mellomnavn: null,
                etternavn: 'Tøffel'
            },
            vergesakstype: 'Fremtidsfullmakt',
            omfang: 'Ivareta personens interesser innenfor det økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-10' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'Norge'
            },
            identifikator: '99009900',
            sistEndret: {
                ident: '11223344',
                tidspunkt: '2018-06-14T00:00:00' as LocalDateTime,
                system: 'BD06'
            },
            prioritet: -1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '12345678',
            sistOppdatert: '2021-11-02' as LocalDate,
            sistVerifisert: '2021-11-02' as LocalDate
        }
    },
    bankkonto: {
        kontonummer: '12345678900',
        banknavn: 'DNB ASA',
        sistEndret: {
            ident: '1010800 BD03',
            tidspunkt: '2006-03-15T00:00:00' as LocalDateTime,
            system: ''
        },
        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: hentBarnAremark()
};
