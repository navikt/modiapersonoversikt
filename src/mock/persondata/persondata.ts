import { erGyldigFødselsnummer } from 'nav-faker/dist/personidentifikator/helpers/fodselsnummer-utils';
import {
    AdresseBeskyttelse,
    Data as PersonData,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    InformasjonElement,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';
import { harDiskresjonskode } from '../../app/personside/visittkort-v2/visittkort-utils';
import { aremark } from './aremark';

// Til bruk under testing av funksjonalitet
const erDod = false;
const visEtiketter = true;
const erReservert = false;
const ikkeRegistrert = false;

export function hentPersondata(fodselsnummer: string): PersonData | null {
    if (fodselsnummer === aremark.personIdent) {
        return { feilendeSystemer: [], person: aremark };
    } else if (fodselsnummer === '05117608222') {
        return {
            feilendeSystemer: [
                InformasjonElement.DKIF,
                InformasjonElement.BANKKONTO,
                InformasjonElement.PDL_GT,
                InformasjonElement.NORG_NAVKONTOR,
                InformasjonElement.NORG_KONTAKTINFORMASJON,
                InformasjonElement.VEILEDER_ROLLER,
                InformasjonElement.PDL_TREDJEPARTSPERSONER
            ],
            person: lagPerson(fodselsnummer)
        };
    } else if (fodselsnummer === '15068215851') {
        return {
            feilendeSystemer: [InformasjonElement.EGEN_ANSATT, InformasjonElement.BANKKONTO],
            person: lagPerson(fodselsnummer)
        };
    } else if (!erGyldigFødselsnummer(fodselsnummer)) {
        return null;
    } else {
        return { feilendeSystemer: [], person: lagPerson(fodselsnummer) };
    }
}

export function lagPerson(fnr: string): Person {
    return {
        personIdent: fnr,
        navn: [
            {
                fornavn: 'TESTFAMILIE',
                mellomnavn: 'MELLOMNAVN',
                etternavn: 'ETTERNAVN'
            }
        ],
        kjonn: [
            {
                kode: Kjonn.K,
                beskrivelse: 'Kvinne'
            }
        ],
        fodselsdato: ['2000-02-02' as LocalDate],
        geografiskTilknytning: '1234',
        alder: 40,
        dodsdato: erDod ? ['2018-06-07' as LocalDate] : [],
        bostedAdresse: [
            {
                linje1: 'Adressevei 1',
                linje2: '0000 AREMARK',
                linje3: 'Norge',
                sistEndret: {
                    ident: 'Folkeregisteret',
                    tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                    system: 'Folkeregisteret'
                },
                gyldighetsPeriode: {
                    gyldigFraOgMed: '2021-02-01' as LocalDate,
                    gyldigTilOgMed: null
                }
            }
        ],
        kontaktAdresse: [
            {
                linje1: 'Kontaktadresse 1',
                linje2: '0320 HEI',
                linje3: null,
                sistEndret: {
                    ident: 'D159000',
                    tidspunkt: '2021-10-10T10:15:30' as LocalDateTime,
                    system: 'NAV'
                },
                gyldighetsPeriode: null
            }
        ],
        oppholdsAdresse: [
            {
                linje1: 'Oppholdsadresse 1',
                linje2: '0000 AREMARK',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: {
                    gyldigFraOgMed: '2021-02-01' as LocalDate,
                    gyldigTilOgMed: '2022-05-01' as LocalDate
                }
            }
        ],
        navEnhet: {
            id: '0219',
            navn: 'NAV Bærum',
            publikumsmottak: [
                {
                    besoksadresse: {
                        linje1: 'Adressevei 1',
                        linje2: '0000 AREMARK',
                        linje3: null,
                        sistEndret: null,
                        gyldighetsPeriode: null
                    },
                    apningstider: [
                        {
                            ukedag: 'Mandag',
                            apningstid: '09.00 - 15.00'
                        },
                        {
                            ukedag: 'Tirsdag',
                            apningstid: '09.00 - 15.00'
                        },
                        {
                            ukedag: 'Onsdag',
                            apningstid: '09.00 - 14.00'
                        }
                    ]
                },
                {
                    besoksadresse: {
                        linje1: 'Adressevei 1',
                        linje2: '0000 AREMARK',
                        linje3: null,
                        sistEndret: null,
                        gyldighetsPeriode: null
                    },
                    apningstider: [
                        {
                            ukedag: 'Mandag',
                            apningstid: '09.00 - 15.00'
                        }
                    ]
                }
            ]
        },
        statsborgerskap: [
            {
                land: {
                    kode: 'NOR',
                    beskrivelse: 'NORGE'
                },
                gyldighetsPeriode: {
                    gyldigFraOgMed: '2021-02-01' as LocalDate,
                    gyldigTilOgMed: '2022-05-01' as LocalDate
                }
            },
            {
                land: {
                    kode: 'DNK',
                    beskrivelse: 'DANMARK'
                },
                gyldighetsPeriode: {
                    gyldigFraOgMed: '2021-02-01' as LocalDate,
                    gyldigTilOgMed: null
                }
            },
            {
                land: {
                    kode: 'FIN',
                    beskrivelse: 'FINLAND'
                },
                gyldighetsPeriode: {
                    gyldigFraOgMed: '2021-02-01' as LocalDate,
                    gyldigTilOgMed: '2022-05-01' as LocalDate
                }
            }
        ],
        adressebeskyttelse: [
            {
                kode: visEtiketter ? AdresseBeskyttelse.KODE6 : AdresseBeskyttelse.UGRADERT,
                beskrivelse: visEtiketter ? 'Sperret adresse, strengt fortrolig' : 'UGRADERT'
            }
        ],
        sikkerhetstiltak: visEtiketter
            ? [
                  {
                      type: 'TFUS',
                      beskrivelse: 'Telefonisk utestengelse',
                      gyldighetsPeriode: {
                          gyldigFraOgMed: '2021-02-01' as LocalDate,
                          gyldigTilOgMed: visEtiketter ? ('2030-02-15' as LocalDate) : ('2010-02-15' as LocalDate)
                      }
                  }
              ]
            : [],
        erEgenAnsatt: visEtiketter ? EgenAnsatt.JA : EgenAnsatt.UKJENT,
        personstatus: [
            {
                kode: erDod ? PersonStatus.DOD : PersonStatus.BOSATT,
                beskrivelse: erDod ? 'Død' : 'Bosatt i Norge'
            }
        ],
        sivilstand: [
            {
                type: {
                    kode: SivilstandType.GIFT,
                    beskrivelse: 'Gift'
                },
                gyldigFraOgMed: '2010-03-06' as LocalDate,
                sivilstandRelasjon: {
                    fnr: '',
                    navn: [],
                    alder: null,
                    adressebeskyttelse: [
                        {
                            kode: AdresseBeskyttelse.KODE6,
                            beskrivelse: 'Sperret adresse, strengt fortrolig'
                        }
                    ],
                    harSammeAdresse: false,
                    dodsdato: []
                }
            }
        ],
        foreldreansvar: forelderBarnMock
            .filter((relasjon) => relasjon.rolle === ForelderBarnRelasjonRolle.BARN)
            .map((barn) => ({
                ansvar: 'felles',
                ansvarlig: null,
                ansvarsubject: {
                    navn: harDiskresjonskode(barn.adressebeskyttelse) ? null : barn.navn.firstOrNull(),
                    ident: barn.ident
                }
            })),
        deltBosted: [
            {
                gyldighetsPeriode: {
                    gyldigFraOgMed: '2021-02-01' as LocalDate,
                    gyldigTilOgMed: '2022-05-01' as LocalDate
                },
                adresse: {
                    linje1: `Adresseveien 1`,
                    linje2: '0000 Aremark',
                    linje3: null,
                    sistEndret: null,
                    gyldighetsPeriode: null
                }
            }
        ],
        dodsbo: visEtiketter
            ? [
                  {
                      adressat: {
                          advokatSomAdressat: null,
                          organisasjonSomAdressat: null,
                          personSomAdressat: {
                              fnr: '10108000398',
                              navn: [
                                  {
                                      fornavn: '',
                                      mellomnavn: null,
                                      etternavn: ''
                                  }
                              ],
                              fodselsdato: '10.10.1980' as LocalDate
                          }
                      },
                      adresse: {
                          linje1: 'Elgelia 20',
                          linje2: '0000 Aremark',
                          linje3: null,
                          sistEndret: null,
                          gyldighetsPeriode: null
                      },
                      registrert: '2010-02-02' as LocalDate,
                      skifteform: Skifteform.OFFENTLIG,
                      sistEndret: {
                          ident: 'Folkeregisteret',
                          tidspunkt: '2015-01-01T10:15:30' as LocalDateTime,
                          system: 'Folkeregisteret'
                      }
                  }
              ]
            : [],
        fullmakt: visEtiketter
            ? [
                  {
                      motpartsPersonident: '123456789',
                      motpartsPersonNavn: {
                          fornavn: 'Navn',
                          mellomnavn: null,
                          etternavn: 'Navnesen'
                      },
                      motpartsRolle: FullmaktsRolle.FULLMEKTIG,
                      omrade: [
                          {
                              kode: '*',
                              beskrivelse: ''
                          }
                      ],
                      gyldighetsPeriode: {
                          gyldigFraOgMed: '2021-02-01' as LocalDate,
                          gyldigTilOgMed: '2022-05-01' as LocalDate
                      },
                      digitalKontaktinformasjonTredjepartsperson: {
                          mobiltelefonnummer: '90909090',
                          reservasjon: 'false'
                      }
                  }
              ]
            : [],
        vergemal: visEtiketter
            ? [
                  {
                      ident: '21042900076',
                      navn: null,
                      vergesakstype: 'Voksen',
                      omfang: 'Ivareta personens interesser innenfor det personlige og økonomiske området',
                      embete: 'Fylkesmannen i Troms og Finnmark',
                      gyldighetsPeriode: {
                          gyldigFraOgMed: '2021-02-01' as LocalDate,
                          gyldigTilOgMed: '2022-05-01' as LocalDate
                      }
                  },
                  {
                      ident: '123456799',
                      navn: {
                          fornavn: 'Truls',
                          mellomnavn: null,
                          etternavn: 'Tøffel'
                      },
                      vergesakstype: 'Fremtidsfullmakt',
                      omfang: 'Ivareta personens interesser innenfor det økonomiske området',
                      embete: 'Fylkesmannen i Troms og Finnmark',
                      gyldighetsPeriode: {
                          gyldigFraOgMed: '2021-02-01' as LocalDate,
                          gyldigTilOgMed: null
                      }
                  }
              ]
            : [],
        tilrettelagtKommunikasjon: {
            talesprak: visEtiketter
                ? [
                      {
                          kode: 'NO',
                          beskrivelse: 'Norsk'
                      }
                  ]
                : [],
            tegnsprak: visEtiketter
                ? [
                      {
                          kode: 'NO',
                          beskrivelse: 'Norsk'
                      }
                  ]
                : []
        },
        telefonnummer: [
            {
                retningsnummer: {
                    kode: '+47',
                    beskrivelse: 'Norge'
                },
                identifikator: '99009900',
                sistEndret: {
                    ident: '11223344',
                    tidspunkt: '2018-06-01T00:00:00' as LocalDateTime,
                    system: 'BD06'
                },
                prioritet: 1
            },
            {
                retningsnummer: {
                    kode: '+47',
                    beskrivelse: 'Norge'
                },
                identifikator: '55003399',
                sistEndret: {
                    ident: '11223344',
                    tidspunkt: '2011-06-14T00:00:00' as LocalDateTime,
                    system: ''
                },
                prioritet: -1
            }
        ],
        kontaktOgReservasjon: ikkeRegistrert
            ? {
                  personident: '10108000398',
                  reservasjon: erReservert ? 'true' : 'false',
                  epostadresse: {
                      value: null,
                      sistOppdatert: null,
                      sistVerifisert: null
                  },
                  mobiltelefonnummer: {
                      value: null,
                      sistOppdatert: null,
                      sistVerifisert: null
                  }
              }
            : {
                  personident: '10108000398',
                  reservasjon: erReservert ? 'true' : 'false',
                  epostadresse: {
                      value: 'epost@nav.no',
                      sistOppdatert: '2013-01-01' as LocalDate,
                      sistVerifisert: '2013-01-01' as LocalDate
                  },
                  mobiltelefonnummer: {
                      value: '90000000',
                      sistOppdatert: '2015-02-01' as LocalDate,
                      sistVerifisert: null
                  }
              },
        bankkonto: {
            kontonummer: '12345678910',
            banknavn: 'DNB ASA',
            sistEndret: {
                ident: '1010800 BD03',
                tidspunkt: '2006-03-15T00:00:00' as LocalDateTime,
                system: ''
            },
            bankkode: null,
            swift: 'DNBANOKKXXX',
            landkode: null,
            adresse: {
                linje1: 'Bankveien 1,',
                linje2: '0357 Bankestad',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            },
            valuta: {
                kode: 'NOK',
                beskrivelse: 'Norske kroner'
            }
        },
        forelderBarnRelasjon: forelderBarnMock
    };
}

const forelderBarnMock: ForelderBarnRelasjon[] = [
    {
        ident: '12345678910',
        rolle: ForelderBarnRelasjonRolle.MOR,
        navn: [
            {
                fornavn: 'MOR',
                mellomnavn: null,
                etternavn: 'MORSAN'
            }
        ],
        fodselsdato: ['1971-03-15' as LocalDate],
        alder: 50,
        kjonn: [
            {
                kode: Kjonn.K,
                beskrivelse: 'Kvinne'
            }
        ],
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.UGRADERT,
                beskrivelse: 'Militær'
            }
        ],
        harSammeAdresse: true,
        dodsdato: []
    },
    {
        ident: '12345678911',
        rolle: ForelderBarnRelasjonRolle.BARN,
        navn: [
            {
                fornavn: 'BARN1',
                mellomnavn: null,
                etternavn: 'BARNESEN'
            }
        ],
        fodselsdato: ['2010-04-21' as LocalDate],
        alder: null,
        kjonn: [
            {
                kode: Kjonn.M,
                beskrivelse: 'Mann'
            }
        ],
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.UGRADERT,
                beskrivelse: 'Pendler'
            }
        ],
        harSammeAdresse: false,
        dodsdato: ['2014-01-01' as LocalDate]
    },
    {
        ident: '',
        rolle: ForelderBarnRelasjonRolle.FAR,
        navn: [],
        fodselsdato: [],
        alder: null,
        kjonn: [],
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.KODE6,
                beskrivelse: 'Sperret adresse, strengt fortrolig'
            }
        ],
        harSammeAdresse: false,
        dodsdato: []
    },
    {
        ident: '12345678913',
        rolle: ForelderBarnRelasjonRolle.BARN,
        navn: [
            {
                fornavn: 'BARN4',
                mellomnavn: null,
                etternavn: 'BARNESEN'
            }
        ],
        fodselsdato: ['1998-04-09' as LocalDate],
        alder: 13,
        kjonn: [
            {
                kode: Kjonn.U,
                beskrivelse: 'Ukjent kjønn'
            }
        ],
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.UGRADERT,
                beskrivelse: 'UGRADERT'
            }
        ],
        harSammeAdresse: true,
        dodsdato: []
    },
    {
        ident: '',
        rolle: ForelderBarnRelasjonRolle.BARN,
        navn: [],
        fodselsdato: [],
        alder: null,
        kjonn: [],
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.KODE7,
                beskrivelse: 'Sperret adresse, fortrolig'
            }
        ],
        harSammeAdresse: false,
        dodsdato: []
    }
];
