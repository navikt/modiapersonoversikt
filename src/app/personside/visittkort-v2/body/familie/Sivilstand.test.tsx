import { render } from '@testing-library/react';
import { aremark } from '../../../../../mock/persondata/aremark';
import TestProvider from '../../../../../test/Testprovider';
import SivilstandWrapper from './Sivilstand';

test('viser sivilstand', () => {
    const sivilstand = render(
        <TestProvider>
            <SivilstandWrapper harFeilendeSystem={false} sivilstandListe={aremark.sivilstand} />
        </TestProvider>
    );

    expect(sivilstand.asFragment()).toMatchSnapshot();
});
