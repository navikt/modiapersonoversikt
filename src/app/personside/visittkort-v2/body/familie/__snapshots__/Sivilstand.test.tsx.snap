// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`viser sivilstand 1`] = `
.c1 {
  position: relative;
  display: flex;
  align-items: center;
}

.c2 {
  position: absolute;
  left: -3.125rem;
  width: 3.125rem;
  display: flex;
  justify-content: center;
}

.c2 >svg {
  height: 1.5rem;
  width: auto;
}

.c0 {
  margin: 10px 0 20px 0;
}

.c0:last-child {
  margin: 10px 0 0 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <svg
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 22.4s10.5-8.1 10.5-15.8S13.2-1.7 12 6C10.8-1.7 1.5-1.1 1.5 7.2 1.5 15.4 12 22.4 12 22.4z"
          fill="#C6C2BF"
          stroke="none"
        />
      </svg>
    </div>
    <h4
      className="typo-element"
    >
      Sivilstand
    </h4>
  </div>
  <p
    className="typo-normal"
  >
    Gift
     
    (12.12.2005)
  </p>
  <p
    className="typo-normal"
  >
    BAREMARK TESTFAMILIEN
     (
    DÃ¸d
    )
  </p>
  <p
    className="typo-normal"
  >
    12345555555
  </p>
  <p
    className="typo-normal"
  >
    Bor med bruker
  </p>
</div>
`;
